### PR TITLE
Fix the write_bytes method

### PR DESCRIPTION
**Pull request by vamonte**
#929

### CODE REVIEW

File: parsec/core/fs/workspacefs/workspacefs.py [hunk 1]
```diff
@@ -331,6 +332,7 @@ def get_version_lister(self):
             FSError
         """
         path = FsPath(path)
+        await self.transactions.file_resize(path, offset)
```
Comment (vxgmichel) [inline parsec/core/fs/workspacefs/workspacefs.py]: A negative offset means [end of file](https://github.com/Scille/parsec-cloud/blob/abe8aa0ac64fc05e2f2164b34618c7a8a95b074a/parsec/core/fs/workspacefs/file_transactions.py#L29). In this case, no need to resize.

File: tests/core/fs/workspacefs/test_workspace_fs.py [hunk 2]
```diff
@@ -257,10 +257,15 @@
     assert await alice_workspace.read_bytes("/foo/bar") == b"abcde"
 
     assert await alice_workspace.write_bytes("/foo/bar", b"xyz", offset=1)
-    assert await alice_workspace.read_bytes("/foo/bar") == b"axyze"
-
+    assert await alice_workspace.read_bytes("/foo/bar") == b"axyz"
     assert await alice_workspace.write_bytes("/foo/bar", b"[append]", offset=-1)
-    assert await alice_workspace.read_bytes("/foo/bar") == b"axyze[append]"
+    assert await alice_workspace.read_bytes("/foo/bar") == b"append]"
```
Comment (vxgmichel) [inline tests/core/fs/workspacefs/test_workspace_fs.py]: The expected value is `b"axyze[append]"` (see my previous comment about negative offset)

File: parsec/core/fs/workspacefs/workspacefs.py (hunk 1 above)
Comment (vamonte) [reply parsec/core/fs/workspacefs/workspacefs.py]: C'est fixé.

File: tests/core/fs/workspacefs/test_workspace_fs.py (hunk 2 above)
Comment (vamonte) [reply tests/core/fs/workspacefs/test_workspace_fs.py]: C'est fixé.

File: parsec/core/fs/workspacefs/workspacefs.py [hunk 3]
```diff
@@ -325,13 +326,17 @@ def get_version_lister(self):
...
+        self, path: AnyPath, data: bytes, offset: int = 0, truncate: bool = True
+    ) -> int:
         """
         Raises:
             FSError
         """
         path = FsPath(path)
         _, fd = await self.transactions.file_open(path, "w")
+        if offset >= 0 and truncate:
+            await self.transactions.fd_resize(fd, offset)
```
Comment (vxgmichel) [inline parsec/core/fs/workspacefs/workspacefs.py]: This should go inside the `try-finally` block in order to make sure that the file descriptor gets closed.

File: tests/core/fs/workspacefs/test_workspace_fs.py [hunk 4]
```diff
@@ -257,10 +257,14 @@
...
-
+    assert await alice_workspace.read_bytes("/foo/bar") == b"axyz"
     assert await alice_workspace.write_bytes("/foo/bar", b"[append]", offset=-1)
-    assert await alice_workspace.read_bytes("/foo/bar") == b"axyze[append]"
+    assert await alice_workspace.read_bytes("/foo/bar") == b"axyz[append]"
+
+    assert await alice_workspace.write_bytes("/foo/bar", b"abcde")
+    assert await alice_workspace.read_bytes("/foo/bar") == b"abcde"
+    assert await alice_workspace.write_bytes("/foo/bar", b"xyz", offset=5)
+    assert await alice_workspace.read_bytes("/foo/bar") == b"abcdexyz"
```
Comment (vxgmichel) [inline tests/core/fs/workspacefs/test_workspace_fs.py]: More tests? For instance:
```python

    # Pathlib mode (truncate=True)
    await alice_workspace.write_bytes("/foo/bar", b"abcde")
    assert await alice_workspace.read_bytes("/foo/bar") == b"abcde"
    await alice_workspace.write_bytes("/foo/bar", b"xyz", offset=1)
    assert await alice_workspace.read_bytes("/foo/bar") == b"axyz"
    await alice_workspace.write_bytes("/foo/bar", b"[append]", offset=-1)
    assert await alice_workspace.read_bytes("/foo/bar") == b"axyz[append]"

    # Clear the content of an existing file
    await alice_workspace.write_bytes("/foo/bar", b"")
    assert await alice_workspace.read_bytes("/foo/bar") == b""

    # Atomic write mode (truncate=False)
    assert await alice_workspace.write_bytes("/foo/bar", b"abcde", truncate=False) == 5
    assert await alice_workspace.read_bytes("/foo/bar") == b"abcde"
    assert await alice_workspace.write_bytes("/foo/bar", b"xyz", offset=1, truncate=False) == 3
    assert await alice_workspace.read_bytes("/foo/bar") == b"axyze"
    assert await alice_workspace.write_bytes("/foo/bar", b"[append]", offset=-1, truncate=False) == 8
    assert await alice_workspace.read_bytes("/foo/bar") == b"axyze[append]"
```

**Pull request merged**